pyscriptjs/src/plugin.ts

```diff
@@ -85,10 +100,47 @@ export class PluginManager {
     afterStartup(runtime: Runtime) {
         for (const p of this._plugins)
             p.afterStartup(runtime);
+        for (const p of this._pythonPlugins)
+            if (typeof p.afterStartup !== 'undefined') {
+                p.afterStartup(runtime);
+            }
     }
 
     onUserError(error: UserError) {
         for (const p of this._plugins)
             p.onUserError(error);
     }
 }
+
+
+export function create_custom_element(name: string, pyObject: any, callback?: Function) : any {
+    logger.info(`creating plugin: ${name}`)
+    class CustomElementWrapper extends HTMLElement {
+        shadow: ShadowRoot;
+        wrapper: HTMLElement;
+        pyObject: any;
+        source: string;
+
+        constructor() {
+            logger.debug(`creating ${name} plugin instance`)
+            super();
+
+            this.shadow = this.attachShadow({ mode: 'open' });
+            this.wrapper = document.createElement('slot');
+            this.shadow.appendChild(this.wrapper);
+            this.source = this.innerHTML;
+            this.pyObject = pyObject(this);
+            if (typeof callback !== 'undefined') {
+                callback(this);
+            }
```

Comment:
`x !== undefined` raises a `ReferenceError` if `x` has not been declared, whereas `typeof x !== 'undefined'` allows for the case when no variable named `x` has been declared. 

In this case, `callback` is definitely defined since it's an argument. Since it should either be a function or undefined, I'd probably just go with `if(callback)`.